app/forms/openid_connect_authorize_form.rb

```diff
@@ -62,7 +62,7 @@ def loa3_requested?
 
   def allowed_form_action
     return unless sp_redirect_uri =~ %r{https?://}
-    SecureHeadersWhitelister.extract_domain(sp_redirect_uri)
+    SecureHeadersWhitelister.extract_scheme_and_domain(sp_redirect_uri)
```

Comment:
I believe paths do not matter since this is a redirect.

From the [CSP Spec](https://www.w3.org/TR/CSP2/#source-list-paths-and-redirects):

> To avoid leaking path information cross-origin (as discussed in Egor Homakov’s Using Content-Security-Policy for Evil), the matching algorithm ignores the path component of a source expression if the resource being loaded is the result of a redirect. For example, given a page with an active policy of img-src example.com not-example.com/path: